### PR TITLE
Demo now works on docker network

### PR DESCRIPTION
**Pull request by ross-p-smith**
Added a `make demo-docker` target so we can show the demo working on docker.

We no longer need to `make start-host` when running the demo any longer.

Relates to #146

### CODE REVIEW

File: data-reconciliation-app/Makefile [hunk 1]
```diff
@@ -53,7 +53,13 @@ unit-test: ## 🧪 Run the Application unit-test
...
 demo: ## 🎬 Demo the Data Reconciliation Application
 	@echo -e "\e[34m$@\e[0m" || true
-	@./test/test.sh --nodeAddress 127.0.0.1:8000 --certificate_dir ./workspace/sandbox_common --interactive
+	@. ../scripts/test_sandbox.sh --nodeAddress 127.0.0.1:8000 --certificate_dir ./workspace/sandbox_common --constitution_dir ./governance/constitution --interactive
+
+# Run `make start-host` in another terminal before running this.
+# To build and start a ccf network and automatically deploy your application on it.
+demo-docker: ## 🎬 Demo the Data Reconciliation Application
+	@echo -e "\e[34m$@\e[0m" || true
+	@. ../scripts/test_docker.sh --virtual --serverIP 172.17.0.3 --port 8080 --interactive
```
Comment (julioalex-rezende) [inline data-reconciliation-app/Makefile:62]: comment seems out-of-date

File: scripts/test_docker.sh [hunk 2]
```diff
@@ -6,19 +6,20 @@ declare enclave_type=""
 declare app_dir=$PWD                   # application folder for reference
 declare app_name=${app_dir##*/}        # application name (to be used in container commands)
 declare certificate_dir="${app_dir}/workspace/docker_certificates"
+declare interactive=0
 
 function usage {
     echo ""
     echo "Start a CCF node in docker and run the tests."
     echo ""
     echo "usage: ./test_docker.sh --serverIP <IPADDRESS> --port <PORT> [--virtual] [--enclave]"
```
Comment (julioalex-rezende) [inline scripts/test_docker.sh:15]:
```suggestion
    echo "usage: ./test_docker.sh --serverIP <IPADDRESS> --port <PORT> [--virtual] [--enclave] [--interactive]"
```

File: scripts/test_docker.sh [hunk 3]
```diff
@@ -61,6 +63,13 @@ fi
 declare server="https://${serverIP}:${port}"
 
 function finish {
+    if [ $interactive -eq 1 ]; then
+        echo "🤔 Do you want to stop the container? (Y/n)"
+        read -r proceed
+        if [ "$proceed" == "n" ]; then
+            exit 0
+        fi
+    fi
```
Comment (julioalex-rezende) [inline scripts/test_docker.sh:72]: If one does not stop the container to keep interacting with the network, how do they stop it later on?
I suppose we'll lose track of it's id and will end up with a dangling network.

File: scripts/test_sandbox.sh [hunk 4]
```diff
@@ -15,8 +16,8 @@ function usage {
     echo "  --nodeAddress       string      The IP and port of the primary CCF node"
     echo "  --certificate_dir   string      The directory where the certificates are"
     echo "  --constitution_dir  string      The directory where the constitution is"
+    echo "  --interactive       boolean     Optional. Run in Demo mode"
```
Comment (julioalex-rezende) [inline scripts/test_sandbox.sh:19]: Please include switch on usage description (line 14)

File: data-reconciliation-app/docs/demo-guidance.md [hunk 5]
```diff
@@ -2,30 +2,214 @@
...
+📂 Working directory (for certificates): ./workspace/sandbox_common
+💤 Waiting for the app frontend...
+```
 
-7. Member 0 submits proposal of code change to the network​ to expose some additional data
-   - We will demo adding a another field (`total_vote_count`) to our reporting object. `total_vote_count` is the number of members who submitted data on that data record.
-8. Member 1 & 2 approve the proposal​
-9. Member 1 views updated report with additional data (`total_vote_count`) added by the proposal
+### Demo notes
+This part of the demo has started the network and deployed the app. The network is running with 3 members, and the app is deployed with the constitution defined [here](../governance/constitution/), which means that all members have equal votes on decisions, and unanimous approval is required to advance proposals. All members have been activated.
```
Comment (julioalex-rezende) [inline data-reconciliation-app/docs/demo-guidance.md:25]:
```suggestion
This part of the demo has started the network and deployed the app. The network is running with 3 members and 1 user, and the app is deployed with the constitution defined [here](../governance/constitution/), which means that all members have equal votes on decisions, and a majority of approval votes is required to advance proposals. All members have been activated.
```

File: data-reconciliation-app/docs/demo-guidance.md [hunk 6]
```diff
@@ -2,30 +2,214 @@
...
+  "members_in_agreement": 2,
+  "lei": "984500UF3DE41EFA7F02",
+  "nace": "B.08.9"
+}
+
+🎬 Full Reports Complete
+
+- Press any key to continue...
+```
+
```
Comment (julioalex-rezende) [inline data-reconciliation-app/docs/demo-guidance.md:117]: nit:

For consistency, `## Demo notes` title is missing

File: data-reconciliation-app/docs/demo-guidance.md [hunk 7]
```diff
@@ -2,30 +2,214 @@
...
+  }
+}
+🎬 LACK_OF_CONSENSUS DATA
+
+- Press any key to continue...
+```
+
+## Part 7. Ingesting more data
+
+To show how we can gain consensus, we can ingest more data and then re-run the report.
```
Comment (julioalex-rezende) [inline data-reconciliation-app/docs/demo-guidance.md:188]:
```suggestion
To show how data ingestion can affect all members' reports, we can ingest more data and then re-run the report. We demonstrate this here with `member0` ingesting some new data and `member2` asking for a report for the same key as in the previous step. You can see that grou_status has changed from `LACK_OF_CONSENSUS` to `IN_CONSENSUS`.
```

File: scripts/test_docker.sh [hunk 8]
```diff
@@ -62,6 +64,14 @@ declare server="https://${serverIP}:${port}"
 
 function finish {
     containerId=$(docker ps -qf ancestor="$app_name:$enclave_type")
+    if [ $interactive -eq 1 ]; then
+        echo "🤔 Do you want to stop the container? (Y/n)"
+        read -r proceed
+        if [ "$proceed" == "n" ]; then
+            echo "👍 Container will continue to run. (Container ID: ${containerId})"
```
Comment (julioalex-rezende) [inline scripts/test_docker.sh:71]:
```suggestion
            echo "👍 Container will continue to run. (Container ID: ${containerId}). Remember to stop it manually when finished."
```

File: scripts/test_sandbox.sh [hunk 9]
```diff
@@ -66,8 +68,15 @@ sandbox_pid=$!
 echo "💤 Waiting for sandbox . . . (${sandbox_pid})"
 
 function finish {
-  kill -9 $sandbox_pid
-  echo "💀 Killed process ${sandbox_pid}"
+    if [ $interactive -eq 1 ]; then
+        echo "🤔 Do you want to stop the sandbox (${sandbox_pid})? (Y/n)"
+        read -r proceed
+        if [ "$proceed" == "n" ]; then
+            exit 0
```
Comment (julioalex-rezende) [inline scripts/test_sandbox.sh:75]: should we include the same warning related to PID in sandbox mode?

**Review by julioalex-rezende — COMMENTED** [review]
Up until Part 3, `Demo notes` were added after the examples. From Part 4 onwards, the notes are in the beginning of each section.
Please put all in the same standard. IMHO, having the notes at the beginning (as we have in Part 4) provided a better readability, so I would go for it:
- Explanation
- Example with the Demo Output

File: data-reconciliation-app/docs/demo-guidance.md (hunk 6 above)
Comment (ross-p-smith) [reply data-reconciliation-app/docs/demo-guidance.md:117]: Removed Demo notes

File: scripts/test_sandbox.sh (hunk 9 above)
Comment (ross-p-smith) [reply scripts/test_sandbox.sh:76]: Done

**Pull request merged**